pkg/externalauthproviders/flags.go

```diff
@@ -0,0 +1,351 @@
+			Question: "Claim mapping groups",
+			Default:  result.claimMappingGroupsClaim,
+			Help:     cmd.Lookup("claim-mapping-groups-claim").Usage,
+			Required: true,
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if interactive.Enabled() && !cmd.Changed("claim-validation-rule") {
+		claimValidationRuleInput, err := interactive.GetString(interactive.Input{
+			Question: "Claim validation rule",
+			Default:  strings.Join(claimValidationRuleSlice, ","),
+			Help:     cmd.Lookup("claim-validation-rule").Usage,
+			Validators: []interactive.Validator{
+				ocm.ValidateClaimValidationRules,
+			},
+		})
+		if err != nil {
+			return nil, err
+		}
+		claimValidationRuleSlice = helper.HandleEmptyStringOnSlice(strings.Split(claimValidationRuleInput, ","))
+		result.claimValidationRule = claimValidationRuleSlice
+	}
+
+	if interactive.Enabled() && !cmd.Changed("console-client-id") {
+		result.consoleClientId, err = interactive.GetString(interactive.Input{
+			Question: "Console client id",
+			Default:  result.consoleClientId,
+			Help:     cmd.Lookup("console-client-id").Usage,
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if interactive.Enabled() && !cmd.Changed("console-client-secret") {
+		if result.consoleClientId != "" {
+			// skips if no consoleClientId is provided
+			result.consoleClientSecret, err = interactive.GetString(interactive.Input{
+				Question: "Console client secret",
+				Default:  result.consoleClientSecret,
+				Help:     cmd.Lookup("console-client-secret").Usage,
+			})
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	return result, nil
+}
+
+func CreateExternalAuthConfig(args *ExternalAuthProvidersArgs) (*cmv1.ExternalAuth, error) {
+	externalAuthBuilder := cmv1.NewExternalAuth().ID(args.name)
+	claimValidationRules := args.claimValidationRule
+
+	if args.issuerUrl != "" || args.issuerCaFile != "" || args.issuerAudiences != nil {
+		tokenIssuerBuilder := cmv1.NewTokenIssuer()
+
+		if args.issuerUrl != "" {
+			tokenIssuerBuilder.URL(args.issuerUrl)
+		}
+
+		if args.issuerCaFile != "" {
+			// Get certificate contents
+			ca := ""
+			if args.issuerCaFile != "" {
+				cert, err := os.ReadFile(args.issuerCaFile)
+				if err != nil {
+					return &cmv1.ExternalAuth{}, fmt.Errorf("expected a valid certificate bundle: %s", err)
+				}
+				ca = string(cert)
+			}
+			// Set the CA file, if any
+			if ca != "" {
+				tokenIssuerBuilder.CA(ca)
+			}
+		}
+
+		if args.issuerAudiences != nil {
+			tokenIssuerBuilder.Audiences(args.issuerAudiences...)
+		}
+		externalAuthBuilder.Issuer(tokenIssuerBuilder)
+	}
+
+	if args.claimMappingGroupsClaim != "" || args.claimMappingUsernameClaim != "" || claimValidationRules != nil {
+		groupClaimBuilder := cmv1.NewGroupsClaim().Claim(args.claimMappingGroupsClaim)
+		usernameClaimBuilder := cmv1.NewUsernameClaim().Claim(args.claimMappingUsernameClaim)
+
+		// there's a bug here that username is not created in the backend. Not sure if it's a bug
```

Comment:
Will remove the comment once the latest api-model -> sdk is dumped. There was a casing issue in the backend.

Reply:
Removed